pyiron_atomistics/atomistics/structure/structurestorage.py

```diff
@@ -175,6 +178,16 @@ def add_structure(self, structure, identifier=None, **arrays):
             **kwargs: additional arrays to store for structure
         """
 
+        # Extract energies, forces, and atomic charges from ASE Atoms objects.
+        if isinstance(structure, ASEAtoms):
+            arrays['charges'] = structure.get_initial_charges()
+
+            if structure.calc is not None:
+                arrays['energy'] = structure.calc.get_potential_energy()
+                arrays['forces'] = structure.calc.get_forces()
```

Comment:
Have you checked that this gives the right array shapes and such for `charges`, `energy` and `forces`?  `FlattenedStorage` should do this automatically, but it's best to check.